### PR TITLE
Add link capability

### LINKED ISSUE #14: Add capability to manage links
Add capability to be able to 
* Link issues
* Get issue link
* Delete issue link

API reference: https://docs.atlassian.com/software/jira/docs/api/REST/7.6.1/#api/2/issueLink

### PR DESCRIPTION
**Pull request by mateiidavid**
Fix #14 

* Add link capability in the form of 3 different commands that can be used: `Create`, `Get` and `Delete`.
* Added some TODOs to refactor some code (will tackle in separate PR).
* The way testing is done in most commands is not necessarily feasible and we should look at improving, most of the times we are just testing that the request is sent but not that the body of the request (or the path) have been properly created.

### CODE REVIEW

File: command/issue_link_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"testing"
+)
+
+func TestLinkCreateIsSuccessful(t *testing.T) {
+	initialFunc := client.SendRequestWithoutResp
+	defer func() { client.SendRequestWithoutResp = initialFunc }()
+	in := []byte(`
+    {
+	"inwardIssue": "TEST-123",
+    "outwardIssue": "Test-321",
```
Comment (YToutah993) [inline command/issue_link_test.go:21]: Minor formatting issue.

Comment (mateiidavid) [reply command/issue_link_test.go:21]: Good spot!

**Comment by YToutah993** [comment]
@mateiidavid QQ please. When I need to link an issue to another one, do I need to provide a linkType? Thank you!

File: command/issue_link_test.go (hunk 1 above)
Comment (mateiidavid) [reply command/issue_link_test.go]: Done!

**Comment by mateiidavid** [comment]
> @mateiidavid QQ please. When I need to link an issue to another one, do I need to provide a linkType? Thank you!

Yup! This is in accordance to the actual REST API specs for Jira 🤔 I think the main reasoning here is that there are different types of links, e.g we can use a "Blocks" or a "Depends" and each project has specific types that it is allowed to do.

**Review by YToutah993 — APPROVED** [review]

**Review by SuganJoe — APPROVED** [review]

**Review by rneumaier — APPROVED** [review]

**Pull request merged**